src/TrezorKeyring.ts

```diff
@@ -234,35 +283,46 @@ class TrezorKeyring extends EventEmitter {
    * Accepts either an ethereumjs-tx or @ethereumjs/tx transaction, and returns
    * the same type.
    *
-   * @template {TypedTransaction | OldEthJsTransaction} Transaction
-   * @param {string} address - Hex string address.
-   * @param {Transaction} tx - Instance of either new-style or old-style ethereumjs transaction.
-   * @returns {Promise<Transaction>} The signed transaction, an instance of either new-style or old-style
+   * @param address - Hex string address.
+   * @param tx - Instance of either new-style or old-style ethereumjs transaction.
+   * @returns The signed transaction, an instance of either new-style or old-style
    * ethereumjs transaction.
    */
-  signTransaction(address, tx) {
+  async signTransaction(
+    address: string,
+    tx: TypedTransaction | OldEthJsTransaction,
+  ) {
     if (isOldStyleEthereumjsTx(tx)) {
       // In this version of ethereumjs-tx we must add the chainId in hex format
       // to the initial v value. The chainId must be included in the serialized
       // transaction which is only communicated to ethereumjs-tx in this
       // value. In newer versions the chainId is communicated via the 'Common'
       // object.
-      return this._signTransaction(address, tx.getChainId(), tx, (payload) => {
-        tx.v = Buffer.from(payload.v, 'hex');
-        tx.r = Buffer.from(payload.r, 'hex');
-        tx.s = Buffer.from(payload.s, 'hex');
-        return tx;
-      });
+      return this.#signTransaction(
+        address,
+        // @types/ethereumjs-tx and old ethereumjs-tx versions document
+        // this function return value as Buffer, but the actual
+        // Transaction._chainId will always be a number.
+        // See https://github.com/ethereumjs/ethereumjs-tx/blob/v1.3.7/index.js#L126
+        tx.getChainId() as unknown as number,
+        tx,
+        (payload) => {
+          tx.v = Buffer.from(payload.v, 'hex');
+          tx.r = Buffer.from(payload.r, 'hex');
+          tx.s = Buffer.from(payload.s, 'hex');
+          return tx;
+        },
+      );
     }
-    return this._signTransaction(
+    return this.#signTransaction(
       address,
       Number(tx.common.chainId()),
       tx,
       (payload) => {
         // Because tx will be immutable, first get a plain javascript object that
         // represents the transaction. Using txData here as it aligns with the
         // nomenclature of ethereumjs/tx.
-        const txData = tx.toJSON();
+        const txData = tx.toJSON() as TxData;
```

Comment:
Since the return type of `toJSON` here is compatible with `TxData`, we don't need to use `as`. Better to avoid `as` if possible because it can be used for downcasting, meaning if we break the types here, the `as` might hide that from us.

If we format it like this, it still works and it still checks to ensure the types are compatible:
```suggestion
        const txData: TxData = tx.toJSON();
```